app/controllers/spree/checkout_controller_decorator.rb

```diff
@@ -0,0 +1,48 @@
+module Spree::CheckoutControllerDecorator
+  def self.prepended(base)
+    base.before_action :checkout_afterpay_payment, only: [:update]
+  end
+
+  # Afterpay Specific Methods Start
+  def checkout_afterpay_payment
+    if afterpay_payment_present?
+      @order.checkout_afterpay_payments.map(&:invalidate!)
+      create_afterpay_payment
+
+      # Remove other payment method parameters.
+      params[:order].delete(:payments_attributes)
+      params.delete(:payment_source)
+      afterpay_source = @order.checkout_afterpay_payments.last.try(:source)
+      if afterpay_source && afterpay_source.checkout(@order.outstanding_balance, currency: current_currency, auto_capture: true)
+        # Always send `auto_capture` as true to Afterpay because we cannot send `capture` event from Spree to Afterpay.
+        # Set the completed_at field of the current order, so that it is no longer shown to the user in the current session
+        @order.update_columns(completed_at: Time.current, special_instructions: Spree.t(:order_marked_paid_info, scope: :afterpay, time: Time.current, number: @order.number))
```

Comment:
You are not resetting special_instructions.

Also, why are we setting this info? special_instructions is normally used for shipment carriers